sympy/integrals/transforms.py

```diff
@@ -1137,6 +1190,607 @@ def sbs(expr):
+    for trigrule in trigrules:
+        fm, nu, s1, s2, sd = trigrule
+        ma1 = func.match(fm*z)
+        if ma1:
+            ma2 = ma1[y].collect(t).match(a*t)
+            if ma2:
+                debug('_laplace_apply_rules match:')
+                debug('      f:    %s ( %s, %s )'%(f, ma1, ma2))
+                debug('      rule: multiply with %s (%s)'%(fm.func, nu))
+                L = _laplace_apply_rules(ma1[z], t, s, doit=doit, **hints)
+                try:
+                    r, p, c = L
+                    # The convergence plane changes only if the shift has been
+                    # done along the real axis:
+                    if sd==1:
+                        cp_shift = Abs(ma2[a])
+                    else:
+                        cp_shift = 0
+                    return ((s1*(r.subs(s, s-sd*ma2[a])+\
+                                    s2*r.subs(s, s+sd*ma2[a]))).simplify()/2,
+                            p+cp_shift, c)
+                except TypeError:
+                    if doit==True and _simplify==True:
+                        return (s1*(L.subs(s, s-sd*ma2[a])+\
+                                    s2*L.subs(s, s+sd*ma2[a]))).simplify()/2
+                    else:
+                        return (s1*(L.subs(s, s-sd*ma2[a])+\
+                                    s2*L.subs(s, s+sd*ma2[a])))/2
+    return None
+
+def _laplace_rule_diff(f, t, s, doit=True, **hints):
+    """
+    This function tries to transform an expression containing a derivative
+    of an undefined function and returns `None` if it cannot do it.
+    """
+    hints.pop('simplify', True)
+    a = Wild('a', exclude=[t])
+    y = Wild('y')
+    n = Wild('n', exclude=[t])
+    g = WildFunction('g', nargs=1)
+    ma1 = f.match(a*Derivative(g, (t, n)))
+    if ma1 and ma1[g].args[0] == t and ma1[n].is_integer:
+        debug('_laplace_apply_rules match:')
+        debug('      f:    %s'%(f,))
+        debug('      rule: time derivative (1.11, 1.12)')
+        d = []
+        for k in range(ma1[n]):
+            if k==0:
+                y = ma1[g].func(t).subs(t, 0)
+            else:
+                y = Derivative(ma1[g].func(t), (t, k)).subs(t, 0)
+            d.append(s**(ma1[n]-k-1)*y)
+        r = s**ma1[n]*_laplace_apply_rules(ma1[g].func(t), t, s, doit=doit,
+                                           **hints)
+        return r - Add(*d)
+    return None
+
+
+def _laplace_apply_rules(f, t, s, doit=True, **hints):
+    """
+    Helper function for the class LaplaceTransform.
+
+    This function does a Laplace transform based on rules and, after
+    applying the rules, hands the rest over to `_laplace_transform`, which
+    will attempt to integrate.
+
+    If it is called with `doit=False`, then it will instead return
+    `LaplaceTransform` objects.
+    """
+
+    k, func = f.as_independent(t, as_Add=False)
+
+    simple_rules = _laplace_build_rules(t, s)
+    for t_dom, s_dom, check, plane, prep in simple_rules:
```

Comment:
Maybe this part could also be factored out into a function and then listed with the other functions in `prog_rules`

Reply:
I thought about it, but I would like to keep it as is, because if we  unify it for all integral transforms later and decide on a way how to allow users to supply their own rules to the function, it will lead to a re-design anyway.